components/brave_wallet_ui/components/buy-send-swap/swap-input-component/index.tsx

```diff
@@ -139,22 +145,28 @@ function SwapInputComponent (props: Props) {
       {componentType !== 'selector' &&
         <>
           <Row>
-            <FromBalanceText isExchange={componentType === 'exchange'}>{getTitle()}</FromBalanceText>
-            {componentType === 'exchange' ? (
+            <FromBalanceText componentType={componentType}>{getTitle()}</FromBalanceText>
+            {componentType === 'exchange' &&
               <MarketLimitButton onClick={onToggleOrderType}>{orderType === 'market' ? locale.swapLimit : locale.swapMarket}</MarketLimitButton>
-            ) : (
+            }
+            {componentType !== 'exchange' && componentType !== 'toAddress' &&
               <FromBalanceText>{locale.balance}: {selectedAssetBalance}</FromBalanceText>
-            )
+            }
+            {componentType === 'toAddress' &&
+              <PasteButton onClick={onPaste}>
+                <PasteIcon />
+              </PasteButton>
             }
           </Row>
-          <Row isExchange={componentType === 'exchange'}>
+          <Row componentType={componentType}>
             <Input
-              isExchange={componentType === 'exchange'}
-              type='number'
-              placeholder='0.0'
-              value={selectedAssetInputAmount}
+              componentType={componentType}
+              type={componentType === 'toAddress' ? 'text' : 'number'}
+              placeholder={componentType === 'toAddress' ? '0x address or url' : '0.0'}
```

Comment:
Nit: Is the placeholder localized correctly for regions where it's `0,0`? Otherwise, it's probably easier to just use `0` here.

Reply:
Has not been setup for localization, but you raise a valid point.  Let's chat about how you went about this in the current wallet extension! 🙏  Changed to `0` for now!